src/views/Login.vue

```diff
@@ -153,7 +154,19 @@ export default {
       await this.setLoginLoading(true);
       await this.setLoginError(null);
       try {
-        const { user } = await auth.signInWithPopup(loginProvider);
+        const { user } = await auth.signInWithPopup(loginProviderGoogle);
+        this.$toasted.show(this.$t('toaster.welcome', { user: user.displayName ? user.displayName : '' }));
+      } catch (e) {
+        await this.setLoginError(2);
+      }
+      await this.setLoginLoading(false);
+    },
+
+    async loginWithMicrosoft() {
+      await this.setLoginLoading(true);
+      await this.setLoginError(null);
+      try {
+        const { user } = await auth.signInWithRedirect(loginProviderMS);
         this.$toasted.show(this.$t('toaster.welcome', { user: user.displayName ? user.displayName : '' }));
```

Comment:
usikker om Microsoft returnerer et objekt som har property `displayName`, så denne kan fort bli tom hvis den ikke gjør det. Får selv vurdere om det er greit nok eller ikke

Reply:
Tenker det ikke er så alt for farlig.